src/renderer/dx/DxRenderer.cpp

```diff
@@ -1168,10 +1178,12 @@ try
     D2D1_COLOR_F nothing = { 0 };
 
     // If the entire thing is invalid, just use one big clear operation.
-    // This will also hit the gutters outside the usual paintable area.
-    // Invalidating everything is supposed to happen with resizes of the
-    // entire canvas, changes of the font, and other such adjustments.
-    if (_invalidMap.all())
+    // Don't use _invalidMap.all() here. That method checks if the union of the
+    // invalid areas of the map is the entire frame. However, there's a good
+    // chance that _not_ all the cells of the frame are actually invalid. If we
+    // used all() here, we'd clear the entire frame, but not actually paint all
+    // the cells again.
+    if (_firstFrame)

```

Comment:
Go change `til::bitmap::all()` to call and return `_bits.all()` instead of what it's doing with the rectangle. That'll do it instead of this flagging.

Reply:
will the dynamic_bitset changes eventually obviate the need to cache the dirty region rect?

Reply:
right now we still need it for `bitmap::one()` for VT

Reply:
we cna use dynamic_bitset.one though, right?

Reply:
oh we can use `dynamic_bitset.count() == 1`, which is cheap w/ libpopcnt